waiter/test/waiter/reporters_test.clj

```diff
@@ -0,0 +1,173 @@
+;;
+;;  http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+(ns waiter.reporters-test
+  (:require [clj-time.core :as t]
+            [clojure.test :refer :all]
+            [metrics.core :as mc]
+            [metrics.counters :as counters]
+            [waiter.metrics :refer :all]
+            [waiter.reporter :as r])
+  (:import (clojure.lang ExceptionInfo)
+           (com.codahale.metrics MetricFilter MetricRegistry ConsoleReporter)
+           (com.codahale.metrics.graphite GraphiteReporter GraphiteSender)
+           (java.io PrintStream ByteArrayOutputStream)))
+
+(def ^:private all-metrics-match-filter (reify MetricFilter (matches [_ _ _] true)))
+
+(defmacro with-isolated-registry
+  [& body]
+  `(with-redefs [mc/default-registry (MetricRegistry.)]
+     (.removeMatching mc/default-registry all-metrics-match-filter)
+     (do ~@body)
+     (.removeMatching mc/default-registry all-metrics-match-filter)))
+
+(deftest console-reporter-bad-schema
+  (is (thrown-with-msg? ExceptionInfo #"period-ms missing-required-key"
+                        (r/validate-console-reporter-config {:extra-key 444}))))
+(deftest console-reporter-good-schema
+  (r/validate-console-reporter-config {:period-ms 300 :extra-key 444}))
+
+(deftest graphite-reporter-bad-schema
+  (is (thrown-with-msg? ExceptionInfo #"host missing-required-key"
+                        (r/validate-graphite-reporter-config {:extra-key 444}))))
+(deftest graphite-reporter-bad-schema-2
+  (is (thrown-with-msg? ExceptionInfo #":port \(not \(pos\?"
+                        (r/validate-graphite-reporter-config {:period-ms 300 :host "localhost" :port -7777}))))
+(deftest graphite-reporter-bad-schema-3
+  (is (thrown-with-msg? ExceptionInfo #":period-ms \(not \(integer\?"
+                        (r/validate-graphite-reporter-config {:period-ms "five" :host "localhost" :port 7777}))))
+(deftest graphite-reporter-good-schema
+  (r/validate-graphite-reporter-config {:period-ms 300 :extra-key 444 :host "localhost" :port 7777}))
+
+(defn make-printstream []
+  (let [os (ByteArrayOutputStream.)
+        ps (PrintStream. os)]
+    {:ps ps :out #(.toString os "UTF8")}))
+
+(deftest console-reporter-no-filter
+  (with-isolated-registry
+    (service-counter "service-id" "foo")
+    (service-counter "service-id" "foo" "bar")
+    (service-counter "service-id" "fee" "fie")
+    (counters/inc! (service-counter "service-id" "foo" "bar") 100)
+    (let [{:keys [ps out]} (make-printstream)
+          [console-reporter state] (r/make-console-reporter nil ps)]
+      (is (instance? ConsoleReporter console-reporter))
+      (.report console-reporter)
+      (is (= "
```

Comment:
Nitpick: Exact string equality seems suspect, is there another alternate way to express this assertion?